upgrade from drs-1 to drs-3 fails

panic: UnmarshalJSON cannot decode empty bytes

goroutine 638 [running]:
github.com/cosmos/cosmos-sdk/x/params/types.Subspace.Get({{0x44c1cc0, 0xc0009d84c0}, 0xc000b02018, {0x4475ae8, 0xc001c25c00}, {0x4475b10, 0xc001c25cf0}, {0xc000993ae0, 0xd, 0x19}, ...}, ...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cosmos-sdk@v0.46.17-0.20241128210616-e9dfe47b8c73/x/params/types/subspace.go:110 +0x287
github.com/cosmos/cosmos-sdk/x/params/types.Subspace.GetParamSet({{0x44c1cc0, 0xc0009d84c0}, 0xc000b02018, {0x4475ae8, 0xc001c25c00}, {0x4475b10, 0xc001c25cf0}, {0xc000993ae0, 0xd, 0x19}, ...}, ...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cosmos-sdk@v0.46.17-0.20241128210616-e9dfe47b8c73/x/params/types/subspace.go:239 +0x105
github.com/dymensionxyz/dymension-rdk/x/rollappparams/keeper.Keeper.GetParams(...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymension-rdk@v1.6.1-0.20241216135050-77cdd63f1653/x/rollappparams/keeper/keeper.go:31
github.com/dymensionxyz/rollapp-wasm/app/upgrades/drs-3.CreateUpgradeHandler.func1({{0x44a0060, 0x6681600}, {0x44c1be0, 0xc0071ade40}, {{0xb, 0x1}, {0xc0009df4b8, 0x17}, 0x2a2, {0x22a22fd1, ...}, ...}, ...}, ...)
	/home/sergi/workspace/dymension/rollapp-wasm/app/upgrades/drs-3/upgrade.go:18 +0x168
github.com/cosmos/cosmos-sdk/x/upgrade/keeper.Keeper.ApplyUpgrade({{0x7fff8bd4221f, 0x1b}, 0xc000baad50, {0x4475ae8, 0xc001c25c20}, {0x44c1cc0, 0xc0009d84c0}, 0xc00170cf60, {0x44663a0, 0xc0020cf340}, ...}, ...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cosmos-sdk@v0.46.17-0.20241128210616-e9dfe47b8c73/x/upgrade/keeper/keeper.go:337 +0x1a3
github.com/cosmos/cosmos-sdk/x/upgrade.BeginBlocker({{0x7fff8bd4221f, 0x1b}, 0xc000baad50, {0x4475ae8, 0xc001c25c20}, {0x44c1cc0, 0xc0009d84c0}, 0xc00170cf60, {0x44663a0, 0xc0020cf340}, ...}, ...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cosmos-sdk@v0.46.17-0.20241128210616-e9dfe47b8c73/x/upgrade/abci.go:62 +0x64d
github.com/cosmos/cosmos-sdk/x/upgrade.AppModule.BeginBlock(...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cosmos-sdk@v0.46.17-0.20241128210616-e9dfe47b8c73/x/upgrade/module.go:134
github.com/cosmos/cosmos-sdk/types/module.(*Manager).BeginBlock(_, {{0x44a0060, 0x6681600}, {0x44c1be0, 0xc0071ade40}, {{0xb, 0x1}, {0xc0009df4b8, 0x17}, 0x2a2, ...}, ...}, ...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cosmos-sdk@v0.46.17-0.20241128210616-e9dfe47b8c73/types/module/module.go:484 +0x1b5
github.com/dymensionxyz/rollapp-wasm/app.(*App).BeginBlocker(_, {{0x44a0060, 0x6681600}, {0x44c1be0, 0xc0071ade40}, {{0xb, 0x1}, {0xc0009df4b8, 0x17}, 0x2a2, ...}, ...}, ...)
	/home/sergi/workspace/dymension/rollapp-wasm/app/app.go:961 +0x13b
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).BeginBlock(_, {{0xc00259fdc0, 0x20, 0x20}, {{0xb, 0x1}, {0xc0009df4b8, 0x17}, 0x2a2, {0x22a22fd1, ...}, ...}, ...})
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cosmos-sdk@v0.46.17-0.20241128210616-e9dfe47b8c73/baseapp/abci.go:185 +0x878
github.com/tendermint/tendermint/abci/client.(*localClient).BeginBlockSync(_, {{0xc00259fdc0, 0x20, 0x20}, {{0xb, 0x1}, {0xc0009df4b8, 0x17}, 0x2a2, {0x22a22fd1, ...}, ...}, ...})
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cometbft@v0.34.29-0.20241128205759-2a9d5f015da5/abci/client/local_client.go:280 +0x103
github.com/tendermint/tendermint/proxy.(*appConnConsensus).BeginBlockSync(_, {{0xc00259fdc0, 0x20, 0x20}, {{0xb, 0x1}, {0xc0009df4b8, 0x17}, 0x2a2, {0x22a22fd1, ...}, ...}, ...})
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/cometbft@v0.34.29-0.20241128205759-2a9d5f015da5/proxy/app_conn.go:81 +0x55
github.com/dymensionxyz/dymint/block.(*Executor).ExecuteBlock(0xc000443d00, 0xc0025fe248)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc01.0.20241213113333-2a07ba36444a/block/executor.go:254 +0x357
github.com/dymensionxyz/dymint/block.(*Manager).applyBlock(0xc0022284e0, 0xc0025fe248, 0xc000715ef0, {0x1, 0x0, {0x0, 0x0, 0x0}})
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc01.0.20241213113333-2a07ba36444a/block/block.go:87 +0x50f
github.com/dymensionxyz/dymint/block.(*Manager).produceApplyGossip(0xc0022284e0, {0x44a08b0, 0xc002828be0}, {0x0?, 0x0?, 0x0?})
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc01.0.20241213113333-2a07ba36444a/block/produce.go:143 +0x1ac
github.com/dymensionxyz/dymint/block.(*Manager).ProduceApplyGossipBlock(...)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc01.0.20241213113333-2a07ba36444a/block/produce.go:118
github.com/dymensionxyz/dymint/block.(*Manager).ProduceBlockLoop(0xc0022284e0, {0x44a08b0, 0xc002828be0}, 0xc00282b110)
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc01.0.20241213113333-2a07ba36444a/block/produce.go:52 +0x236
github.com/dymensionxyz/dymint/block.(*Manager).runAsProposer.func2()
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc01.0.20241213113333-2a07ba36444a/block/modes.go:98 +0x65
github.com/dymensionxyz/dymint/block.(*Manager).runAsProposer.ErrGroupGoLog.func6()
	/home/sergi/go/pkg/mod/github.com/dymensionxyz/dymint@v1.2.0-rc01.0.20241213113333-2a07ba36444a/utils/errors/err_group.go:18 +0x30
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/sergi/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 182
	/home/sergi/go/pkg/mod/golang.org/x/sync@v0.8.0/errgroup/errgroup.go:75 +0x96

--- a/app/upgrades/drs-3/upgrade.go
+++ b/app/upgrades/drs-3/upgrade.go
@@ -15,7 +15,7 @@ func CreateUpgradeHandler(
 	configurator module.Configurator,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
-		if rpKeeper.GetParams(ctx).DrsVersion == 1 {
+		if rpKeeper.Version(ctx) == 1 {
 			// first run drs-2 migration
 			if err := drs2.HandleUpgrade(ctx, rpKeeper); err != nil {
 				return nil, err

